### PR TITLE
Java prerequisites

### PR DESCRIPTION
**Pull request by bentley-ritson-umuzi**
Added java.md file and updated dependencies in java projects.

### CODE REVIEW

File: content/syllabuses/java.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+- Java Generics
+- {{% contentlink "topics/java-specific/generics" %}}
+- {{% contentlink "projects/java-specific/generics" %}}
+- Spring Infrastructure 
+- {{% contentlink "topics/java-specific/introduction-to-spring-infrastructure/part-1" %}}
+- {{% contentlink "topics/java-specific/introduction-to-spring-infrastructure/part-2" %}}
+- {{% contentlink "topics/java-specific/introduction-to-spring-infrastructure/part-3" %}}
+- {{% contentlink "projects/java-specific/introduction-to-spring-infrastructure/part-1" %}}
+- {{% contentlink "projects/java-specific/introduction-to-spring-infrastructure/part-2" %}}
+- {{% contentlink "projects/java-specific/introduction-to-spring-infrastructure/part-3" %}}
```
Comment (sheenarbw) [inline content/syllabuses/java.md:20]: doesn't part 3 rely on part 2? And doesn't part 2 rely on part 1?  What is the goal that the recruits are aiming for? Should all this stuff have a card that is highlighted?

File: content/syllabuses/java.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+- Java Collections
+- {{% contentlink "topics/java-specific/collections-and-datastructures" %}}
+- {{% contentlink "projects/java-specific/collections" %}}
+- Java Data Structures
+- {{% contentlink "topics/java-specific/collections-and-datastructures" %}}
+- {{% contentlink "projects/java-specific/data-structures" %}}
+- Java Generics
+- {{% contentlink "topics/java-specific/generics" %}}
+- {{% contentlink "projects/java-specific/generics" %}}
```
Comment (sheenarbw) [inline content/syllabuses/java.md:13]: Doesnt the project have the topic as a prerequisite? 
I would remove this line: 
- {{% contentlink "topics/java-specific/generics" %}}

File: content/syllabuses/java.md (hunk 1 above)
Comment (bentley-ritson-umuzi) [reply content/syllabuses/java.md:12]: Implemented.

File: content/syllabuses/java.md (hunk 2 above)
Comment (bentley-ritson-umuzi) [reply content/syllabuses/java.md:10]: implemented across all.

**Pull request merged**